pyanaconda/modules/payload/sources/live_os.py

```
@@ -0,0 +1,137 @@
+            except FileNotFoundError:
+                pass
+
+        # Is it a squashfs+overlayfs base image?
+        if os.path.exists("/run/rootfsbase"):
+            try:
+                block_device = execWithCapture("findmnt",
+                                               ["-n", "-o", "SOURCE", "/run/rootfsbase"]).strip()
+                if block_device:
+                    log.debug("Detected live base image %s", block_device)
+                    return block_device
+            except (OSError, FileNotFoundError):
+                pass
+
+        log.debug("No live base image detected")
+        return ""
+
+    def set_up_with_tasks(self):
+        """Set up the installation source for installation.
+
+        :return: list of tasks required for the source setup
+        :rtype: [Task]
+        """
+        task = SetUpInstallationSourceTask(self._image_path, INSTALL_TREE)
+
+        task.succeeded_signal.connect(lambda: self._set_is_ready(True))
+
+        return [task]
+
+    def tear_down_with_tasks(self):
+        """Tear down the installation source for installation.
+
+        :return: list of tasks required for the source clean-up
+        :rtype: [Task]
+        """
+        task = TearDownInstallationSourceTask(INSTALL_TREE)
+
+        task.succeeded_signal.connect(lambda: self._set_is_ready(False))
+
+        return [task]
+
+    def validate(self):
+        """Test if the image exists on the given path.
+
+        :return: True if file on the path exists.
+        """
+        try:
+            res = stat.S_ISBLK(os.stat(self._image_path)[stat.ST_MODE])
+            if res:
+                log.debug("Live OS source is valid %s", self._image_path)
+                return True
+            else:
+                log.warning("Live OS source is not valid %s", self._image_path)
+        except FileNotFoundError:
+            log.warning("Live OS source is not available %s", self._image_path)
+
+        return False
```

This is not a good enough information. We should really propagate the error messages. Also, I would be more happy to see this code in a task even if it is run locally with `run()`.

---

Fixed by removing this method.